### ISSUE TITLE
InconsistentKeys error being returned where DuplicateKeys is expected

### ISSUE THREAD

**Issue by LAshenhurst** [digital-asset/daml#9457]
Creating this issue as requested in forum thread: https://discuss.daml.com/t/2481/

When repeatedly exercising a choice with the below logic, an `InconsistentKeys` error is returned by the ledger API, whereas it seems like the correct error would be `DuplicateKeys`.

Please note, repeated commands are sent with a delay to ensure there is no race. There are no concurrent transactions when this error is produced.

```
template Vendor
 with
   networkOwner: Party
   vendor: Party
   vendorId: Text
 where
   signatory networkOwner
   observer vendor

   key (networkOwner, vendorId): (Party, Text)
   maintainer key._1

   controller networkOwner can
     nonconsuming RegisterProduct: (Optional (ContractId Client), ContractId Product)
       with productInfo: ProductInformation; clientInfo: ClientInformation
       do
         clientCid <- lookupByKey @Client (clientKey)
         productCid <- lookupByKey @Product (productKey)

         newClient <- case clientCid of
            None -> do
                exerciseByKey @ClientList (clientListKey) AddClients with newClients = [clientInfo]
                case clientInfo.clientType of
                  FIRSTTYPE -> do 
                      newClient <- create Client with ...
                      return (Some newClient)
                  SECONDTYPE -> do return None
            Some clientCid -> do
                client <- fetch clientCid
                archive clientCid
                newClient <- create client with products = dedup (productInfo :: client.products) 
                return (Some newClient)
 
        product <- create Product with ...

        return (newClient, product)
```

**Comment by fabiotudone-da** [comment]
@LAshenhurst thanks, I can indeed reproduce it in sandbox+navigator by adding the same product to the same client twice.

**Comment by fabiotudone-da** [comment]
@LAshenhurst It should have been fixed as part of #9478 and indeed I cannot reproduce it with the current SDK `main`. Could you try as well and confirm before we close this issue? A new snapshot build should be out fairly soon that includes the fix or, if you feel like it, you could also build/install the SDK yourself from `main`.

**Comment by fabiotudone-da** [comment]
@LAshenhurst daml `1.13.0-snapshot.20210426.6770.0.ca66061b` is now available.

**Comment by cocreature** [comment]
Hi @LAshenhurst, have you been able to verify that this fixes your issue? On our side, we are unable to reproduce this issue with the fix in 1.13.0 or anything later.

**Comment by LAshenhurst** [comment]
Hi @cocreature apologies for the delayed reply. Yes, this seems resolved now. Thanks for following up!
